input/collector.go

```diff
@@ -41,33 +40,49 @@ type TCPCollector struct {
 // UDPCollector represents a network collector that accepts UDP packets.
 type UDPCollector struct {
 	addr   *net.UDPAddr
-	parser *RFC5424Parser
+	parser types.Parser
+}
+
+func (s *TCPCollector) Addr() net.Addr {
+	return s.addr
 }
 
 // NewCollector returns a network collector of the specified type, that will bind
 // to the given inteface on Start(). If config is non-nil, a secure Collector will
 // be returned. Secure Collectors require the protocol be TCP.
-func NewCollector(proto, iface string, tlsConfig *tls.Config) Collector {
-	parser := NewRFC5424Parser()
+func NewCollector(proto, input string, iface string, tlsConfig *tls.Config) types.Collector {
```

Comment:
I suggest you do the introduction of the interface first, and we review and merge that first.